src/invidious/helpers/jobs.cr

```diff
@@ -3,15 +3,17 @@ def refresh_channels(db, logger, config)
 
   spawn do
     max_threads = max_channel.receive
+    lim_threads = max_threads
     active_threads = 0
     active_channel = Channel(Bool).new
+    backoff = 1.seconds
 
     loop do
       db.query("SELECT id FROM channels ORDER BY updated") do |rs|
         rs.each do
           id = rs.read(String)
 
-          if active_threads >= max_threads
+          if active_threads >= lim_threads
```

Comment:
If the idea of this is to start at max threads (say 5) and if an error occurs, force the job to drop to one thread, I think this needs to be changed from an if to a while loop. There's a lot going on in this so I could be wrong.

Reply:
I think you are correct